scripts/one-off/schedules-migration/migrate.ts

```diff
@@ -0,0 +1,159 @@
+                    schedule.status,
+                    schedule.sync_id,
+                    schedule.frequency,
+                    schedule.created_at,
+                    schedule.updated_at,
+                    schedule.deleted_at,
+                    sync.name as sync_name,
+                    connection.id as nango_connection_id,
+                    connection.connection_id as connection_id,
+                    connection.provider_config_key as provider_config_key,
+                    connection.environment_id as environment_id
+                `)
+        )
+        .from<SourceSchedule>({ schedule: `nango._nango_sync_schedules` })
+        .join('nango._nango_syncs as sync', 'schedule.sync_id', 'sync.id')
+        .join('nango._nango_connections as connection', 'sync.nango_connection_id', 'connection.id')
+        .where({ 'schedule.deleted_at': null })
+        .orderBy('schedule.created_at', 'asc');
+}
+
+async function writeSchedule(schedule: SourceSchedule): Promise<{ id: string }> {
+    const frequencyMs = ms(schedule.frequency);
+    if (!frequencyMs) {
+        return Promise.reject(new Error(`Invalid frequency: ${schedule.frequency}`));
+    }
+    const targetSchedule: TargetSchedule = {
+        id: uuidv7(),
+        name: `environment:${schedule.environment_id}:sync:${schedule.sync_id}`,
+        state: schedule.status === 'PAUSED' ? 'PAUSED' : 'STARTED',
+        starts_at: new Date(schedule.created_at),
+        frequency: `${ms(schedule.frequency)} milliseconds`,
+        payload: {
+            type: 'sync',
+            syncId: schedule.sync_id,
+            syncName: schedule.sync_name,
+            debug: false,
+            connection: {
+                id: schedule.nango_connection_id,
+                connection_id: schedule.connection_id,
+                provider_config_key: schedule.provider_config_key,
+                environment_id: schedule.environment_id
+            }
+        },
+        group_key: 'sync',
+        retry_max: 0,
+        created_to_started_timeout_secs: 3600,
+        started_to_completed_timeout_secs: 86400,
+        heartbeat_timeout_secs: 1800,
+        created_at: new Date(schedule.created_at),
+        updated_at: new Date(schedule.updated_at),
+        deleted_at: null
+    };
+    return targetKnex.insert(targetSchedule).into<{ id: string }>(`nango_scheduler.schedules`).onConflict(['name']).ignore().returning(['id']);
+}
+
+async function migrate() {
+    console.log('Starting schedules migration...');
+    console.log('Source DB:', SOURCE_DB_URL);
+    console.log('Target DB:', TARGET_DB_URL);
+
+    const schedules = await readSchedules();
+    for (const schedule of schedules) {
+        try {
+            const res = await writeSchedule(schedule);
+            console.log('Schedule migrated:', res);
+        } catch (err) {
+            console.error('Error migrating schedule:', err);
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
waiting so the scheduling worker doesn't have to handle all new schedules at the same time.